src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -1174,7 +1173,7 @@ def create_service_principal_for_rbac(
         existing_sps = list(graph_client.service_principal_list(filter=query_exp))
 
     app_start_date = datetime.datetime.now(datetime.timezone.utc)
-    app_end_date = app_start_date + relativedelta(years=years or 1)
+    app_end_date = app_start_date + datetime.timedelta(days=_years_to_days(years))
```

Comment:
There is no need to use `years or 1` as the fallback logic already happens at L1162:

https://github.com/Azure/azure-cli/blob/6db34d33dcb442e9d01072bfdd34d156afcdd149/src/azure-cli/azure/cli/command_modules/role/custom.py#L1162